components/builder-api/src/depot/server.rs

```diff
@@ -2654,6 +2827,94 @@ fn do_cache_response(response: &mut Response) {
     )));
 }
 
+fn set_memcache_package(
+    client: &mut memcache::Client,
+    origin: String,
+    name: String,
+    version: Option<String>,
+    channel: Option<String>,
+    key: &str,
+) -> Result<()> {
```

Comment:
Probably cleaner to omit Result - the callers should not have to check / react to failure to set or delete cache - we don't want to fail the caller ever on set/delete.